src/Fantomas/CodePrinter.fs

```diff
@@ -127,7 +127,9 @@ and genParsedHashDirective (ParsedHashDirective (h, s, r)) =
         | None -> col sepSpace s printArgument
         <| ctx
 
-    !- "#" -- h +> sepSpace +> printIdent
+    genTriviaFor ParsedHashDirective_ r !- "#" -- h
```

Comment:
You actually want to call `genTriviaFor` for the entire expression as `r` cover all elements.
```fsharp
    !- "#" -- h
    +> sepSpace
    +> printIdent
    |> genTriviaFor ParsedHashDirective_ r
```

Other than that, this looks good. Thanks Mike.

Reply:
Nice, I just pushed the new version